plugins/resources/src/unixfilesystem.cpp

```diff
@@ -83,6 +84,80 @@
 const std::string DEFAULT_VAULT_DIR_MODE( "default_vault_directory_mode_kw" );
 const std::string HIGH_WATER_MARK( "high_water_mark" ); // no longer used
 const std::string REQUIRED_FREE_INODES_FOR_CREATE("required_free_inodes_for_create"); // no longer used
+const std::string HOST_MODE("HOST_MODE");
+
+bool is_detached_mode(irods::plugin_property_map& prop_map) {
+
+    std::string host_mode_str;
+
+    irods::error ret = prop_map.get< std::string >(HOST_MODE, host_mode_str);
+    if (ret.ok()) {
+
+        if (boost::iequals(host_mode_str.c_str(), "detached"))  {
+            return true;
+        }
+    }
+
+    // default is attached mode
+    return false;
+}
+
+irods::error unixfilesystem_start_operation(irods::plugin_property_map& prop_map)
+{
+    using logger = irods::experimental::log;
+
+    irods::error ret = SUCCESS();
+
+    bool detached_mode = is_detached_mode(prop_map);
+
+    if (detached_mode) {
+
+        bool error = false;
+
+        // update host to new host
+        char resource_location[MAX_NAME_LEN];
+        gethostname(resource_location, MAX_NAME_LEN);
```

Comment:
do we have an `irods::` way to do this?   i don't remember, but i thought we were getting away from calling this directly.

Reply:
Still looking for this.

Reply:
Does this need to consider host resolution info (i.e. hosts_config.json in 4-2-stable)?

Reply:
right - i feel like we made an official frontdoor for this type of check/lookup...

Reply:
I added resolve_hostname().